scapy/arch/__init__.py

```diff
@@ -10,72 +10,103 @@
 from __future__ import absolute_import
 import socket
 
+from scapy.compat import orb
+from scapy.config import conf, _set_conf_sockets
 from scapy.consts import LINUX, SOLARIS, WINDOWS, BSD
+from scapy.data import ARPHDR_ETHER, ARPHDR_LOOPBACK, IPV6_ADDR_GLOBAL
 from scapy.error import Scapy_Exception
-from scapy.config import conf, _set_conf_sockets
+from scapy.interfaces import NetworkInterface
 from scapy.pton_ntop import inet_pton, inet_ntop
-from scapy.data import ARPHDR_ETHER, ARPHDR_LOOPBACK, IPV6_ADDR_GLOBAL
-from scapy.compat import orb
 
+# Typing imports
+from scapy.compat import (
+    Any,
+    Optional,
+    Union,
+)
+
+# Note: the typing of this file is heavily ignored because MyPy doesn't allow
+# to import the same function from different files.
+
+# This list only includes imports that are common across all platforms.
+__all__ = [  # noqa: F405
```

Comment:
This was required to please mypy. It also helps understanding how this files work a lot IMHO.